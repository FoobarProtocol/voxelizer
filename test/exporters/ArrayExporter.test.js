import { Volume, ArrayExporter } from '../../src/index.js';
import ndarray from 'ndarray';

test('array export of voxels', () => {
  let array = new Array(8).fill(undefined).map((value, index) => {
    return index % 2 == 0 ? 0 : 1;
  });
  let voxels = new ndarray(array, [2, 2, 2]);
  let volume = new Volume(voxels);
  let exporter = new ArrayExporter();
  exporter.parse(volume, (array) => {
    expect(array).toMatchSnapshot();
  });
});

test('array export of voxels and colors', () => {
  let voxelArray = new Array(8).fill(undefined).map((value, index) => {
    return index % 2 == 0 ? 0 : 1;
  });
  let colorArray = new Array(24).fill(undefined).map((value, index) => {
    return index % 255;
  });
  let voxels = new ndarray(voxelArray, [2, 2, 2]);
  let colors = new ndarray(colorArray, [2, 2, 2, 3]);
  let volume = new Volume(voxels, colors);
  let exporter = new ArrayExporter();
  exporter.parse(volume, (array) => {
    expect(array).toMatchSnapshot();
  });
});

test('array export of empty volume', () => {
  let volume = new Volume();
  let exporter = new ArrayExporter();
  exporter.parse(volume, (array) => {
    expect(array).toMatchSnapshot();
  });
});
