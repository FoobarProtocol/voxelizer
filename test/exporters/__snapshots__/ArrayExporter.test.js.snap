// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`array export of empty volume 1`] = `
Array [
  null,
  null,
]
`;

exports[`array export of voxels 1`] = `
Array [
  Array [
    Array [
      Array [
        0,
        1,
      ],
      Array [
        0,
        1,
      ],
    ],
    Array [
      Array [
        0,
        1,
      ],
      Array [
        0,
        1,
      ],
    ],
  ],
  null,
]
`;

exports[`array export of voxels and colors 1`] = `
Array [
  Array [
    Array [
      Array [
        0,
        1,
      ],
      Array [
        0,
        1,
      ],
    ],
    Array [
      Array [
        0,
        1,
      ],
      Array [
        0,
        1,
      ],
    ],
  ],
  Array [
    Array [
      Array [
        Array [
          0,
          1,
          2,
        ],
        Array [
          3,
          4,
          5,
        ],
      ],
      Array [
        Array [
          6,
          7,
          8,
        ],
        Array [
          9,
          10,
          11,
        ],
      ],
    ],
    Array [
      Array [
        Array [
          12,
          13,
          14,
        ],
        Array [
          15,
          16,
          17,
        ],
      ],
      Array [
        Array [
          18,
          19,
          20,
        ],
        Array [
          21,
          22,
          23,
        ],
      ],
    ],
  ],
]
`;
